### DISCUSSION TITLE
PipeWire Audio Interface

### DISCUSSION THREAD

**Discussion by SubhadeepJasu** [FluidSynth/fluidsynth#976] [General]
Are there any plans to adding native support for pipewire along side the usual ones like pulseaudio, alsa, jack, etc?

I am making a Linux app using fluidsynth which gets distributed as flatpaks. It requires low latency and fair quality audio output in realtime, because the app's focus is on live performance. Pulse audio works but sometimes the user is required to make changes to the limits file to allow high priority. Sometimes even that is not enough. I have recently installed pipewire with a pulseaudio like server (part of pipewire legacy support). Fluidsynth seems to connect to that server and work marvelously with good quality audio and little to no latency. 

But I don't know if going forward the legacy interface would still be a good idea or if fluidsynth will hopefully support pipewire as a driver option.

**Comment by mawe42** [comment]
I don't think there are any active plans to implement a native PipeWire driver.

Personally know very little about PipeWire, but from what I've read, the seamless integration with PulseAudio, ALSA, Gstreamer etc is a major feature. So if the PulseAudio interface of Pipewire works well for you... what advantage would you see in a native Pipewire driver?

**Reply by mawe42** [reply]
> I don't think there are any active plans to implement a native PipeWire driver.

Having said that, I also don't think that there are any concerns that prevent adding a native Pipewire driver. So if you would want to draft a PR, that would probably be very welcome.

**Reply by sykhro** [reply]
I've sent in a small implementation, see #982 

> what advantage would you see in a native Pipewire driver?

More and more distros are picking it up as the new default audio server (e.g. Fedora). Since it's low-latency oob with little to no configuration, I'd say that's handy.

PipeWire offers a drop-in replacement for JACK, in my experience it doesn't always work. In fact, on my machines, pipewire-jack doesn't make a sound with fluidsynth. Upstream will fix it eventually I guess.

**Comment by SubhadeepJasu** [comment]
Update: I read their FAQ a bit more. PipeWire is a low level system. They compare it to XOrg. We don't make applications directly with X lib, but use UI frameworks like Gtk, or QT that work with X lib. They recommend that we use whatever we are used for the time and let pipewire serve as an all compatible backend. Until they come up with further API documentation I think I can rely on alsa or pulse audio to process audio through pipewire. Anyway, thanks for the reply.
